main_app/views.py

```diff
@@ -0,0 +1,37 @@
+from django.shortcuts import render
+from django.views.generic import TemplateView
+
+from .igdb_api import IGDB
+
+
+class MainPageView(TemplateView):
+    template = 'main_page.html'
+
+    def __init__(self, **kwargs):
+        super(MainPageView, self).__init__(**kwargs)
```

Comment:
You don't need to override __ init __ method here. With ClassBased views it's quite rare case.

You Should declare all needed variables in `get_context_data`  function